### ISSUE TITLE
SSSS upload can fail during bootstrap leaving the account in a broken state

### ISSUE THREAD

**Issue by dbkr** [vector-im/riot-web#13216]
We upload public keys and then upload the private ones to SSSS: if the latter fails then we end up in an inconsistent state, especially if we were resetting in which case we have different keys in each place

**Comment by jryans** [comment]
Seems nice to have fixed for release, but not a blocker.

**Comment by non-Jedi** [comment]
Is my reading of this correct that there's no longer any value for debugging in maintaining my account in the broken state, and I should just go ahead and reset cross-signing?

**Comment by jryans** [comment]
> Is my reading of this correct that there's no longer any value for debugging in maintaining my account in the broken state, and I should just go ahead and reset cross-signing?

Yes, I think we've learned all we'll learn from the forensics you and @fladd reported in https://github.com/vector-im/riot-web/issues/12959, so feel free to reset now and file any further issues from there. Thanks for the help testing this! 😄

**Comment by fladd** [comment]
Except that resetting doesn't work... https://github.com/vector-im/riot-web/issues/13212

So no, don't reset yet! Wait for the resetting mechanism to be fixed. Otherwise you end up in yet another broken state.

**Comment by jryans** [comment]
> Except that resetting doesn't work... #13212

This is now fixed on develop, so if you're able to try that, if would be useful to use it and reset for verification.

**Comment by non-Jedi** [comment]
After resetting cross-signing keys, verification was successful between the two devices mentioned in #12959. :)

**Comment by bwindels** [comment]
I guess retrying would be a good start, and if it keeps failing, show a notification somehow in the interface that other devices won't be able to cross-sign (and that you could lose your signing capability altogether if your session gets hosed)

**Comment by bwindels** [comment]
Might be related https://github.com/vector-im/riot-web/issues/13169

**Comment by bwindels** [comment]
Might warrant creating some generic retrying infrastructure (with some backoff) in the js-sdk

We might also want a toast while you're in the broken state.

**Comment by bwindels** [comment]
We're retrying the account data requests up to 5 times now, with an exponential backoff. This should be enough of an improvement for `phase:3`

Leaving this open, as if they'd still fail after 5 retries, we'd still end up in the broken state.
